src/webui/src/components/PackageSidebar/index.jsx

```diff
@@ -0,0 +1,192 @@
+    showAllContributors: false,
+    dependencies: {}
+  };
+
+  constructor(props) {
+    super(props);
+    this.showAllContributors = this.showAllContributors.bind(this);
+  }
+
+  async componentDidMount() {
+    await this.loadPackageData(this.props.packageName);
+  }
+
+  async componentWillReceiveProps(newProps) {
+    if (newProps.packageName !== this.props.packageName) {
+      await this.loadPackageData(newProps.packageName);
+    }
+  }
+
+  async loadPackageData(packageName) {
+    let packageMeta;
+
+    try {
+      packageMeta = (await API.get(`sidebar/${packageName}`)).data;
+    } catch (err) {
+      this.setState({
+        failed: true
+      });
+      return;
+    }
+
+    let lastUpdate = 0;
+    Object.keys(packageMeta._uplinks).forEach((upLinkName) => {
+      const status = packageMeta._uplinks[upLinkName];
+
+      if (status.fetched > lastUpdate) {
+        lastUpdate = status.fetched;
+      }
+    });
+
+    let recentReleases = Object.keys(packageMeta.time).map((version) => {
+      return {
+        version,
+        time: packageMeta.time[version]
+      };
+    });
+    recentReleases = recentReleases.slice(recentReleases.length - 3, recentReleases.length).reverse();
+
+    this.setState({
+      lastUpdate: lastUpdate ? (new Date(lastUpdate)).toLocaleString() : '',
+      recentReleases,
+      author: packageMeta.latest.author,
+      contributors: packageMeta.latest.contributors,
+      dependencies: packageMeta.latest.dependencies,
+      showAllContributors: _.size(packageMeta.latest.contributors) <= 5
+    });
+  }
+
+  showAllContributors() {
+    this.setState({
+      showAllContributors: true
+    });
+  }
+
+  render() {
+    let {author, contributors, recentReleases, lastUpdate, showAllContributors, dependencies} = this.state;
+
+    let uniqueContributors = [];
+    if (contributors) {
+      uniqueContributors = _.filter(_.uniqBy(contributors, (contributor) => contributor.name), (contributor) => {
+        return contributor.name !== _.get(author, 'name');
+      })
+        .slice(0, 5);
+    }
+
+    return (
+      <aside>
+        <Module
+          title="Last Sync"
+          description={lastUpdate}
+          className={classes.releasesModule}
+        >
+          <ul>
+            {recentReleases.length > 0 && recentReleases.map((versionInfo) => {
+              return (
+                <li key={versionInfo.version}>
+                  <span>{versionInfo.version}</span>
+                  <span>{(new Date(versionInfo.time)).toLocaleString()}</span>
+                </li>
+              );
+            })}
+          </ul>
+        </Module>
+
+        <Module
+          title="Maintainers"
+          className={classes.authorsModule}
+        >
+          <ul>
+            {author && <MaintainerInfo title="Author" name={author.name} avatar={author.avatar}/>}
+            {contributors && (showAllContributors ? contributors : uniqueContributors).map((contributor, index) => {
```

Comment:
Let's try to not mix logic with JSX, it's hard to read. I'd encourage more the usage of functions for such logic